pkg/fanal/secret/scanner.go

```diff
@@ -459,31 +459,33 @@ func findLocation(start, end int, content []byte) (int, int, types.Code, string)
 		lineEnd += start
 	}
 
-	match := string(content[start:end])
-	matchLine := string(content[lineStart:lineEnd])
-	if len(matchLine) > 100 {
+	var matchLine string
+	if lineEnd-lineStart > 100 {
 		truncatedLineStart := lo.Ternary(start-30 < 0, 0, start-30)
 		truncatedLineEnd := lo.Ternary(end+20 > len(content), len(content), end+20)
 		matchLine = string(content[truncatedLineStart:truncatedLineEnd])
+	} else {
+		matchLine = string(content[lineStart:lineEnd])
 	}
```

Comment:
I'm not sure, but what do you think about next:

```suggestion
		lineStart = lo.Ternary(start-30 < 0, 0, start-30)
		lineEnd = lo.Ternary(end+20 > len(content), len(content), end+20)
	}
	matchLine = string(content[lineStart:lineEnd])
	endLineNum := startLineNum + bytes.Count(content[lineStart:lineEnd], lineSep)
```

Reply:
it looks better written like this, but the interval used to calculate `endLineNum` seems to be a bit of a problem

The previous code also used this interval `start:end`, the number of lines before the `start` point plus the number of lines between the `start` point and the `end` point to get the position of the end line number looks fine.

```go
// this may be easier to understand but it will bring more calculations
endLineNum := bytes.Count(content[:end], lineSep)

// because we already have the startLineNum, there is no need to recalculate the data of interval content[:start]
endLineNum := startLineNum + bytes.Count(content[start:end], lineSep)
```

consider the case where the interval `start:end` contains newlines, if we use `lineStart:lineEnd` interval to calculate `endLineNum`, the `startLineNum` is always equals  `endLineNum` unless `lineEnd - lineStart > 100`.

this problem is caused by miscalculation of `lineEnd := bytes.Index(content[start:], lineSep)`, It does not take into account the fact that the `start:end` interval contains newlines

maybe this should be another issue to solve instead of modifying it here, what do you think

Reply:
@cnlkl oh, thanks for the details! it's my mistake. 
i messed up `matchLine` and `match`... but the tests were passed... strange.

I meant that we can create `matchLine` after `if`-condition, right?

Reply:
yes, create `matchLine` after if-condition look clearer

Reply:
> yes, create `matchLine` after if-condition look clearer

PR has been updated